planner/core/physical_plans.go

```diff
@@ -991,6 +1010,24 @@ func (ts *PhysicalTableScan) SetIsChildOfIndexLookUp(isIsChildOfIndexLookUp bool
 	ts.isChildOfIndexLookUp = isIsChildOfIndexLookUp
 }
 
+// AddExtraPhysTblIDColumn for partition table.
+// For keepOrder with partition table,
+// we need use partitionHandle to distinct two handles,
+// the `_tidb_rowid` in differenct partition will same in some scenario.
+func (ts *PhysicalTableScan) AddExtraPhysTblIDColumn() bool {
```

Comment:
Would it be possible to add a new struct with the common fields from PhysicalTableScan and PhysicalIndexScan and then use that for composing the two different structs?
Then this code would not need to be duplicated.

Reply:
Tried to reorganize code to avoid duplications